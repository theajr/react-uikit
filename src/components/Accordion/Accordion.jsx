import React from 'react'
import PropTypes from 'prop-types'
import cx from '../../utils/cx'

const AccordionItem = ({ title, children, expanded }) => {
  const classes = cx({ 'uk-open': expanded })
  return (
    <li className={classes}>
      <a className='uk-accordion-title' href='#'>
        {title}
      </a>
      <div className='uk-accordion-content'>{children}</div>
    </li>
  )
}

const Accordion = ({ collapsible, children }) => {
  const ukAccordion = `collapsible: ${collapsible}`
  return <ul uk-accordion={ukAccordion}>{children}</ul>
}

Accordion.propTypes = {
  collapsible: PropTypes.bool
}

Accordion.defaultProps = {
  collapsible: true
}

Accordion.Item = AccordionItem

export default Accordion
