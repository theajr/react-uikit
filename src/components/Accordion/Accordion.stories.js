import React from 'react'

import Accordion from './Accordion'

import { withKnobs, text, boolean, radios } from '@storybook/addon-knobs'

import { withInfo } from '@storybook/addon-info'
import { titleCase } from '../../utils/str'

export default {
  title: 'Accordion',
  decorators: [withKnobs, withInfo({ source: true })]
}

export const Gallery = () => {
  const collapsible = boolean('Collapsible?', true)

  return (
    <Accordion collapsible={collapsible}>
      <Accordion.Item title='Hello' expanded>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptate,
        saepe? Nam minus molestias iusto magni natus. Dignissimos quasi
        laboriosam nemo praesentium consequatur nobis labore cum? Velit
        assumenda autem aliquid maxime!
      </Accordion.Item>
      <Accordion.Item title='Bye' expanded>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptate,
        saepe? Nam minus molestias iusto magni natus. Dignissimos quasi
        laboriosam nemo praesentium consequatur nobis labore cum? Velit
        assumenda autem aliquid maxime!
      </Accordion.Item>
    </Accordion>
  )
}
