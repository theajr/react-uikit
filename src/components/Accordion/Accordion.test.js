import React from 'react'
import { shallow } from 'enzyme'
import Accordion from './Accordion'

describe('<Accordion />', () => {
  test('renders', () => {
    const wrapper = shallow(
      <Accordion collapsible>
        <Accordion.Item title='Hello' expanded>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptate,
          saepe? Nam minus molestias iusto magni natus. Dignissimos quasi
          laboriosam nemo praesentium consequatur nobis labore cum? Velit
          assumenda autem aliquid maxime!
        </Accordion.Item>
        <Accordion.Item title='Bye' expanded>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptate,
          saepe? Nam minus molestias iusto magni natus. Dignissimos quasi
          laboriosam nemo praesentium consequatur nobis labore cum? Velit
          assumenda autem aliquid maxime!
        </Accordion.Item>
      </Accordion>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
