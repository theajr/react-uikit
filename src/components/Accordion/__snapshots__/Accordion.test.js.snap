// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> renders 1`] = `ShallowWrapper {}`;
